cocos/particle/renderer/particle-system-renderer-cpu.ts

```diff
@@ -436,7 +436,6 @@ export default class ParticleSystemRendererCPU extends ParticleSystemRendererBas
                     trailModule.removeParticle(p);
                 }
                 this._particles!.removeAt(i);
```

Comment:
`removeAt` will swap current Particle with the last on at `.length-1`, As a result, after updating all particles backwardly, all particles are valid.